### ISSUE TITLE
Create a better error handler

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/meilisearch-swift#40]
When doing tests the only error I recieve whenever there is an error is the following:

```
failed - The operation couldn’t be completed. (MeiliSearch.MSError error 1.)
```

I have to look at meiliSearch logs to understand the problem.

The error handler should return the full body that meiliSearch provides.

## More precision (by @curquiza)

We should add a basic error handler with these 2 custom exception (or error):
- `MeiliSearchApiException`: MeiliSearch (HTTP) sends back a 4xx or a 5xx. There are most of the SDKs errors.
- `MeiliSearchCommunicationException`: problem with communication, for example, the MeiliSearch instance is unreachable. 

FYI: Even if it's not documented yet, MeiliSearch API has a great error handler, and returns this kind of JSON when an error is found:

<img width="603" alt="Capture d’écran 2020-07-02 à 10 18 21" src="https://user-images.githubusercontent.com/20380692/86334372-6880da00-bc4d-11ea-84ae-82b090394923.png">

It could be great to let the user access `message`, `errorCode`, `errorType` and `errorLink`.

**Comment by ppamorim** [comment]
That's because the current tests where using `XCTFail(error.localizedDescription)` to print the error. This is wrong since not all the error is print by this function. I did replace this with `print(error)` in the PR #54.

**Comment by bidoubiwa** [comment]
fixed by: #145
